Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift

```diff
@@ -311,13 +317,22 @@ public final class PackageInfoMapper: PackageInfoMapping {
             }
         )
 
-        let baseSettings = packageSettings.baseSettings.with(
-            base: packageSettings.baseSettings.base.combine(
-                with: [
-                    "OTHER_SWIFT_FLAGS": ["$(inherited)", "-package-name", packageInfo.name.quotedIfContainsSpaces],
-                ]
+        let baseSettings: XcodeGraph.Settings
+
+        let swiftToolsVersion = try swiftPackageManagerController
+            .getToolsVersion(at: path)
+
+        if swiftToolsVersion >= Version(5, 9, 0) {
```

Comment:
So previous code was wrong on Swift < 5.9?

Reply:
Yes, that's correct. It was not in line with what SPM does: https://github.com/swiftlang/swift-package-manager/blob/740a6c552bf90fd76fdea77ba3334ac5c909ef91/Sources/PackageModel/PackageModel.swift#L136